server/sse.go

```diff
@@ -286,11 +287,11 @@ func (s *SSEServer) Shutdown(ctx context.Context) error {
 	s.mu.RUnlock()
 
 	if srv != nil {
-		s.sessions.Range(func(key, value interface{}) bool {
+		s.sessions.Range(func(key string, value ClientSession) bool {
 			if session, ok := value.(*sseSession); ok {
 				close(session.done)
 			}
-			s.sessions.Delete(key)
+			s.sessions.LoadAndDelete(key)
```

Comment:
I'm not sure deleting sessions is the right thing to do during shutdown for a horizontally-scaled deployment. Clients may still have an active session connection to another instance of the server.

Even if it's a single instance, it would make sense to persist sessions through restarts IMO. 

I'd suggest not deleting here. Even on a single instance with the memory store, the sessions would get garbage collected anyway.